frame/contracts/common/src/lib.rs

```diff
@@ -45,3 +56,62 @@ pub enum RentProjection<BlockNumber> {
 	/// E.g. because the contract accumulated enough funds to offset the rent storage costs.
 	NoEviction,
 }
+
+bitflags! {
+	/// Flags used by a contract to customize exit behaviour.
+	#[derive(Encode, Decode)]
+	pub struct ReturnFlags: u32 {
+		/// If this bit is set all changes made by the contract execution are rolled back.
+		const REVERT = 0x0000_0001;
+	}
+}
+
+/// Output of a contract call or instantiation which ran to completion.
+#[derive(PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+pub struct ExecReturnValue {
+	/// Flags passed along by `seal_return`. Empty when `seal_return` was never called.
+	pub flags: ReturnFlags,
+	/// Buffer passed along by `seal_return`. Empty when `seal_return` was never called.
+	pub data: Vec<u8>,
+}
+
+impl ExecReturnValue {
+	/// We understand the absense of a revert flag as success.
+	pub fn is_success(&self) -> bool {
+		!self.flags.contains(ReturnFlags::REVERT)
+	}
+}
+
+/// Call or instantiate both call into other contracts and pass through errors happening
+/// in those to the caller. This enum is for  the caller to distinguish whether the error
+/// happened during the execution of the callee or in the current execution context.
+#[derive(PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+pub enum ErrorOrigin {
+	/// The error happened in the current exeuction context rather than in the one
+	/// of the contract that is called into.
```

Comment:
```suggestion
	/// Caller error origin.
	///
	/// The error happened in the current exeuction context rather than in the one
	/// of the contract that is called into.
```